rnd/autogpt_builder/src/app/login/page.tsx

```diff
@@ -0,0 +1,168 @@
+"use client";
+import useUser from '@/hooks/useUser';
+import { login, signup } from './actions'
+import { Button } from '@/components/ui/button';
+import { Form, FormControl, FormDescription, FormField, FormItem, FormLabel, FormMessage } from '@/components/ui/form';
+import { useForm } from 'react-hook-form';
+import { Input } from '@/components/ui/input';
+import { z } from "zod"
+import { zodResolver } from "@hookform/resolvers/zod"
+import { PasswordInput } from '@/components/PasswordInput';
+import { FaGoogle, FaGithub, FaDiscord, FaSpinner } from "react-icons/fa";
+import { useState } from 'react';
+import { useSupabase } from '@/components/SupabaseProvider';
+import { useRouter } from 'next/navigation';
+
+const loginFormSchema = z.object({
+  email: z.string().email().min(2).max(64),
+  password: z.string().min(6).max(64),
+})
+
+export default function LoginPage() {
+  const { supabase, isLoading: isSupabaseLoading } = useSupabase();
+  const { user, isLoading: isUserLoading } = useUser();
+  const [feedback, setFeedback] = useState<string | null>(null);
+  const router = useRouter();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const form = useForm<z.infer<typeof loginFormSchema>>({
+    resolver: zodResolver(loginFormSchema),
+    defaultValues: {
+      email: "",
+      password: "",
+    },
+  })
+
+  if (user) {
+    console.log('User exists, redirecting to profile')
+    router.push('/profile')
+  }
+
+  if (isUserLoading || isSupabaseLoading || user) {
+    return (
+      <div className="flex justify-center items-center h-[80vh]">
+        <FaSpinner className="mr-2 h-16 w-16 animate-spin"/>
+      </div>
+    );
+  }
+
+  if (!supabase) {
+    return <div>User accounts are disabled because Supabase client is unavailable</div>
+  }
+
+  async function handleSignInWithProvider(provider: 'google' | 'github' | 'discord') {
+    const { data, error } = await supabase!.auth.signInWithOAuth({
+      provider: provider,
+      options: {
+        redirectTo: `http://localhost:3000/auth/callback`,
```

Comment:
is this right? can this be in an env var?

Reply:
I made it an env var with `http://localhost:3000/auth/callback` as default